src/language-js/utils.js

```diff
@@ -11,6 +11,20 @@ const {
 } = require("../common/util");
 const handleComments = require("./comments");
 
+/**
+ * @typedef {import("estree").BaseNode} BaseNode
+ * @typedef {import("estree").BaseExpression} BaseExpression
+ * @typedef {import("estree").Expression} Expression
+ * @typedef {import("estree").MemberExpression} MemberExpression
+ *
+ * @typedef {object} MemberExpressionNode
+ * @mixes MemberExpression
+ *
+ * @typedef {object} TemplateLiteralNode
+ * @mixes BaseNode
+ * @property {Expression[]} expressions
```

Comment:
I have updated these lines to hopefully resolve the comments by @fisker.

I would like to make these types more specific, with help from `@babel/types` or `@typescript-eslint/types`, but still have not been able to get it to work.

Is it OK to continue with these typedefs for now and consider improving them in the future?

Reply:
I don't think we need `MemberExpression`, why not just  `Node` from all parsers